articles/azure-monitor/app/ilogger.md

```diff
@@ -265,6 +265,12 @@ namespace ConsoleApp
                     // Only Application Insights is registered as a logger provider
                     builder.AddApplicationInsights("<YourInstrumentationKey>");
                 });
+                
+                // To use ILogger in different classes by dependency injection
```

Comment:
This is not specific to ApplicationInsightsProvider. Its a general iLogger/DI concept, so it need not be made part of applicationinsights specific doc.

This might be a good addition for : https://docs.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#non-host-console-app

Reply:
Thank you Cijo for your comments. Ideally, only by adding 
services.AddLogging(builder =>
                {
                    builder.AddApplicationInsights(config.ApplicationInsightsKey);
                });
  ILogger should be available in DI for it to use but it is not logging to the Application Insights. To correct this for application insights we need to explicitly call this proposed solution. 
Whenever you need to create an Azure webjob as a console application, just to use ILogger in the same example would help for correct DI for Application Isnights. Hence proposed it. Please let me know in case of any queries.

Reply:
For webjobs, there is alternate SDKs right? also there is workerservice sdk as well which would satisfy the need for most users.

Reply:
https://github.com/microsoft/ApplicationInsights-dotnet/pull/1643/files once this pr is ready, it is possible to write the ApplicationInsightsLogger in console apps *without* requiring to use DI. We intend to modify the console example to be as follows then:
```
class Program
{
static void Main(string[] args)
{
using var loggerFactory = LoggerFactory.Create(builder =>
{
builder.AddApplicationInsights();
});
var logger = loggerFactory.CreateLogger("mycategory");
logger.LogInformation("sample information message! Food is {food} and price is {price}", "orange", 8.9);
}
}
```

This matches what official ilogger docs suggest as well:  https://docs.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#non-host-console-app

Reply:
@cijothomas  : Sure this code also suffice the requirement. Can I update the PR and  propose this solution ?

Reply:
> @cijothomas : Sure this code also suffice the requirement. Can I update the PR and propose this solution ?

https://github.com/microsoft/ApplicationInsights-dotnet/pull/1643/files must be merged and shipped before we can suggest this approach. Other wise there wouldn't be *any* way to configure the TelemetryConfiguration (other than ikey).

Reply:
(also, this solution won't work on .NET Core 2.1. Since that is out of support already, shouldn't be an issue going forward. But need to wait for applciation insight PR to merge first).